src/audio/drc/ref_sin_fixpt.c

```diff
@@ -0,0 +1,319 @@
+      } else {
+        if (((th - 6588397) & 1073741824) != 0) {
+          th = (th - 6588397) | -1073741824;
+        } else {
+          th = (th - 6588397) & 1073741823;
+        }
+
+        negate = 0U;
+      }
+    } else if (th < -1647099) {                         // -20
+      if (((th + 3294199) & 1073741824) != 0) {
+        z = (th + 3294199) | -1073741824;
+      } else {
+        z = (th + 3294199) & 1073741823;
+      }
+
+      if (z >= -1647099) {
+        th = z;
+        negate = 1U;
+      } else {
+        if (((th + 6588397) & 1073741824) != 0) {
+          th = (th + 6588397) | -1073741824;
+        } else {
+          th = (th + 6588397) & 1073741823;
+        }
+
+        negate = 0U;
+      }
+    } else {
+      negate = 0U;
+    }
+
+    th <<= 9;
+    if ((th & 1073741824) != 0) {
+      z = th | -1073741824;
+    } else {
+      z = th & 1073741823;
+    }
+
+    b_yn = 0;
+    xn = 326016644;
+    xtmp = 326016644;
+    ytmp = 0;
+    for (b_idx = 0; b_idx < 10; b_idx++) {
+      if (z < 0) {                              // negative LookUp table
+        th = z + inpLUT[b_idx];
+        if ((th & 1073741824) != 0) {
+          z = th | -1073741824;
+        } else {
+          z = th & 1073741823;
+        }
+
+        th = xn + ytmp;
+        if ((th & 1073741824) != 0) {
+          xn = th | -1073741824;
+        } else {
+          xn = th & 1073741823;
+        }
+
+        th = b_yn - xtmp;
+        if ((th & 1073741824) != 0) {
+          b_yn = th | -1073741824;
+        } else {
+          b_yn = th & 1073741823;
+        }
+      } else {
+        th = z - inpLUT[b_idx];                 // positive LookUp table
+        if ((th & 1073741824) != 0) {
+          z = th | -1073741824;
+        } else {
+          z = th & 1073741823;
+        }
+
+        th = xn - ytmp;
+        if ((th & 1073741824) != 0) {
+          xn = th | -1073741824;
+        } else {
+          xn = th & 1073741823;
+        }
+
+        th = b_yn + xtmp;
+        if ((th & 1073741824) != 0) {
+          b_yn = th | -1073741824;
+        } else {
+          b_yn = th & 1073741823;
+        }
+      }
+
+      th = xn >> (b_idx + 1);
+      if ((th & 1073741824) != 0) {
+        xtmp = th | -1073741824;
+      } else {
+        xtmp = th & 1073741823;
+      }
+
+      th = b_yn >> (b_idx + 1);
+      if ((th & 1073741824) != 0) {
+        ytmp = th | -1073741824;
+      } else {
+        ytmp = th & 1073741823;
+      }
+    }
+
+    if (negate) {
+      if ((-b_yn & 1073741824) != 0) {
+        cdcSinTh[idx] = -b_yn | -1073741824;
+      } else {
+        cdcSinTh[idx] = -b_yn & 1073741823;
+      }
+    } else {
+      cdcSinTh[idx] = b_yn;
+    }
+  }
+}
+
+/*
+ * function thRadFxp = init_data_fixpt
+ * Arguments    : int32_t thRadFxp[1024]
+ * Return Type  : void
+ * Input Q12.20
+ */
+void init_data_fixpt(int32_t thRadFxp[1024])
+{
+  static const int32_t iv[1024] = { 0, 6433, 12867, 19301, 25735, 32169, 38603,
```

Comment:
this is a pretty big table (consumes 4k bytes) is the whole thing needed?

Reply:
This is input test vector